### PR TITLE
renderTo() now accepts DOM nodes

### LINKED ISSUE #1819: [Usability] Be able to renderTo HTML DOM Element
When playing with the library, it made a lot of sense for me to do this

```typescript
var svg = document.getElementById('demo');
chartTable.renderTo(svg);
```

Should this not be supported?

### PR DESCRIPTION
**Pull request by crmorford**
Closes #1819.

### CODE REVIEW

File: src/components/component.ts [hunk 1]
```diff
@@ -262,12 +262,14 @@ export class Component {
...
    */
-  public renderTo(element: String | d3.Selection<void>): Component {
+  public renderTo(element: String | SVGSVGElement | d3.Selection<void>): Component {
     this.detach();
     if (element != null) {
       let selection: d3.Selection<void>;
       if (typeof(element) === "string") {
         selection = d3.select(<string> element);
+      } else if (element instanceof SVGSVGElement) {
+        selection = d3.select(<SVGSVGElement> element);
```
Comment (bluong) [inline src/components/component.ts]: You can collapse this case and the above case into a single case

```typescript
if (typeof(element) === "string || element instanceof SVGSVGElement)) {
  selection = d3.select(<string | SVGSVGElement> element);
}

File: src/components/component.ts [hunk 2]
```diff
@@ -262,12 +262,14 @@ export class Component {
    * @param {String|d3.Selection} element A selector-string for the <svg>, or a d3 selection containing an <svg>.
    * @returns {Component} The calling Component.
    */
-  public renderTo(element: String | d3.Selection<void>): Component {
+  public renderTo(element: String | SVGSVGElement | d3.Selection<void>): Component {
```
Comment (aicioara) [inline src/components/component.ts]: I prefer `HTMLElement` instead of `SVGSVGElement`

The reason for that is that in 99% of the cases the element is taken by using 
```javascript
document.getElementById("foo") // the type of this is not compatible with SVGSVGElement because it is too generic
```

File: test/components/componentTests.ts [hunk 3]
```diff
@@ -445,6 +445,20 @@ describe("Component behavior", () => {
     svg.remove();
   });
 
+  it("renderTo() only accepts strings, selections containing svgs, and SVG elements", () => {
+    svg.attr("id", "render-to-test");
+    assert.doesNotThrow(() => c.renderTo("#render-to-test"), Error, "accepts strings that identify svgs");
+    assert.doesNotThrow(() => c.renderTo(svg), Error, "accepts selections that contain svgs");
+    assert.doesNotThrow(() => c.renderTo(<any> svg.node()), Error, "accepts svg elements");
```
Comment (aicioara) [inline test/components/componentTests.ts]: instead of svg.node(). use `document.getElementById("render-to-test")`. This + the suggestion above will allow you to not need the `<any>` cast anymore

File: test/components/componentTests.ts [hunk 4]
```diff
@@ -445,6 +445,20 @@ describe("Component behavior", () => {
...
+    let parent = TestMethods.getSVGParent();
+    let div = parent.append("div");
+    // HACKHACK #2614: chai-assert.d.ts has the wrong signature
+    (<any> assert).throws(() => c.renderTo(div), Error, "", "rejects selections that don't contain svgs");
```
Comment (aicioara) [inline test/components/componentTests.ts]: I like this assert. Let's make another one where we check the same thing for `div[0][0]` (basically for the DOM element instead of the selectiob)

File: test/components/componentTests.ts [hunk 5]
```diff
@@ -445,6 +445,20 @@ describe("Component behavior", () => {
...
+    (<any> assert).throws(() => c.renderTo("#not-a-element"), Error, "", "rejects strings that don't correspond to DOM elements");
+    (<any> assert).throws(() => c.renderTo(d3.select(null)), Error, "", "rejects empty d3 selections");
```
Comment (aicioara) [inline test/components/componentTests.ts]: The third argument corresponds to the error the assert is throwing. Let's use it to make sure the expected error is thrown

**Comment by aicioara** [comment]
Rest looks really good. I really like the way you think about edge cases in tests. I can see the QE experience.

**Comment by aicioara** [comment]
Looks like SauceLabs / Travis got flaky. I restarted the test for you.

File: test/components/componentTests.ts (hunk 5 above)
Comment (crmorford) [inline test/components/componentTests.ts]: Does this mean I should find out what error is usually thrown when this happens and copy that error there verbatim?

Comment (aicioara) [inline test/components/componentTests.ts]: Yea. You can do that very easily by writing some garbage above, see what the test is expecting and just paste it there. It's better than leaving it empty

File: plottable.d.ts [hunk 6]
```diff
@@ -1449,7 +1449,7 @@ declare module Plottable {
          * @param {String|d3.Selection} element A selector-string for the <svg>, or a d3 selection containing an <svg>.
          * @returns {Component} The calling Component.
          */
-        renderTo(element: String | d3.Selection<void>): Component;
+        renderTo(element: String | Element | d3.Selection<void>): Component;
```
Comment (aicioara) [inline plottable.d.ts]: Is there a reason why we use Element instead of HTMLElement?

**Comment by aicioara** [comment]
Apart from that it looks good

File: plottable.d.ts (hunk 6 above)
Comment (crmorford) [inline plottable.d.ts]: I had to use Element  instead of HTMLElement because of an issue surfaced by one of the tests you recommended.

document.getElementById("render-to-test") instanceof HTMLElement == false
document.getElementById("render-to-test") instanceof Element == true

Comment (aicioara) [inline plottable.d.ts]: I learned something today. SVGSVGElement are not HTMLElement, but they are Element. Cool

**Comment by jtlan** [comment]
Tested:
* `renderTo(selection containing SVG)`
* `renderTo(invalid selection)`
* `renderTo(document.getElementById("chart"))`
* creating an element from scratch:

  ```Javascript
var svg = document.createElementNS("http://www.w3.org/2000/svg", "svg");
svg.setAttribute("width", "400px");
svg.setAttribute("height", "400px");
document.body.appendChild(svg);
chart.renderTo(svg);
```

**Comment by aicioara** [comment]
@crmorford Please delete branch as well
